services/frontend/src/pages/admin/AdminDashboard.jsx

```diff
@@ -151,34 +48,56 @@ const AdminDashboard = ({ intl }) => {
     [intl]
   );
 
-  // useEffect to run once component is mounted
-  useEffect(() => {
-    Promise.all([
-      getUserCount(),
-      getHiddenUserCount(),
-      getInactiveUserCount(),
-      getExfeederUserCount(),
-      getGrowthRateByMonth(),
-      getGrowthRateByWeek(),
-      getTopFiveCompetencies(),
-      getTopFiveSkills(),
-      getTopFiveDevelopmentalGoals(),
-    ]);
-  }, [
-    getExfeederUserCount,
-    getGrowthRateByMonth,
-    getGrowthRateByWeek,
-    getHiddenUserCount,
-    getInactiveUserCount,
-    getTopFiveCompetencies,
-    getTopFiveDevelopmentalGoals,
-    getTopFiveSkills,
-    getUserCount,
-  ]);
+  const getBackendInfo = useCallback(async () => {
+    try {
+      return await Promise.all([
+        axios.get(`api/stats/count/users`),
+        axios.get(`api/stats/count/hiddenUsers`),
+        axios.get(`api/stats/count/inactiveUsers`),
+        axios.get(`api/stats/count/exFeederUsers`),
+        axios.get(`api/stats/growthRateByMonth`),
+        axios.get(`api/stats/growthRateByWeek`),
+        axios.get(`api/stats/topFiveCompetencies?language=${locale}`),
+        axios.get(`api/stats/topFiveDevelopmentalGoals?language=${locale}`),
+        axios.get(`api/stats/topFiveSkills?language=${locale}`),
+      ]);
```

Comment:
Same as above, 
```suggestion
      const [users, hiddenUsers, inactiveUsers, exFeederUsers, growthRateByMonth, growthRateByWeek, ....] =  await Promise.all([
        axios.get(`api/stats/count/users`),
        axios.get(`api/stats/count/hiddenUsers`),
        axios.get(`api/stats/count/inactiveUsers`),
        axios.get(`api/stats/count/exFeederUsers`),
        axios.get(`api/stats/growthRateByMonth`),
        axios.get(`api/stats/growthRateByWeek`),
        axios.get(`api/stats/topFiveCompetencies?language=${locale}`),
        axios.get(`api/stats/topFiveDevelopmentalGoals?language=${locale}`),
        axios.get(`api/stats/topFiveSkills?language=${locale}`),
      ]);
      
      dispatch(setCountUsers(userCount.data));
      dispatch(setCountHiddenUsers(hiddenUserCount.data));
      dispatch(setCountInactiveUsers(inactiveUserCount.data));
      dispatch(setCountExFeederUsers(exFeederUserCount.data));
      dispatch(setGrowthRateByMonth(growthRateByMonth.data));
      dispatch(setGrowthRateByWeek(growthRateByWeek.data));
```

Reply:
Also feel free to modify the actions in redux, so instead of dispatching 9 seperate times to the redux store, you could do it in one batch, it would clean up the code here and the code in the redux admin slice